src/state-tomo.jl

```diff
@@ -6,8 +6,8 @@ export fit,
        LSStateTomo,
        MLStateTomo
 
-function build_state_predictor(obs::Vector{Matrix{T}}) where T
-    return reduce(vcat,[vec(o)' for o in obs])
+function build_state_predictor(obs::Vector{Matrix{T} where T})
```

Comment:
Should we assume Float64 here?  Or at least a number T<:Number?

Reply:
I think Number. It looks like the input can be a vectorized Hermitian matrix, so likely will have complex entries.

Reply:
maybe Complex{Float64}?

Reply:
I've had issues with Complex{Float64} not accepting things that are Float64. This can be an issue if you create an all zero or identify matrix using the zeros() command (such as for testing). If you look at the type tree, I don't think Float64 is a subtype of Complex, so you have to go up to Number.

Reply:
@dieris what object are you passing this function?  I'm having trouble figuring out what issue you were having with the old syntax.

Reply:
This syntax still seems to be working for me in Julia 1.4.  What error are you getting specifically?

Reply:
no method matching build_state_predictor(::Array{Array{T,2} where T,1})

Reply:
@dieris In the code block you sent, I'm not sure I understand how "obs" is an Array{Array{T,2} where T,1}. Is it because it's originally defined as Matrix{}[]? I don't see where the "where T" part is coming in its declaration.

I'm not sure I'm comfortable with arbitrary type declarations inside of the overall type declaration, but I'm also not sure why it should matter. It seems to me like an issue we might want to fix all the way up the chain.

Reply:
yes, it's because the Matrix element type is not specified. We should probably do  `obs = Matrix{Complex{Float64}}[]` or  `obs = Matrix{Number}[]` if you prefer

Reply:
I think we should write a second method for unspecified types if we want to support them.  It's hard to imagine a real-life scenario where the type wouldn't be a complexF64.  Maybe lower bit value Floats or even integers, but not much else.

Reply:
My concern with that is that float is not a subtype of Complex, so if someone entered a float it would error. What we could do is convert the input to ComplexF64 in earlier code, but then we would have to find every instance of the function.